### PR TITLE
Major refactoring

### PR DESCRIPTION
**Pull request by yuvve**
Complete restructure of the code, with the purpose of increasing readability, decoupling and ease of testing.

The new structure is as follows:
- `classes/`: contains classes that are each responsible for a single feature.
- `adapters/`: contains classes that help communicate between other classes (i.e. fetch from SCB API and insert into DB).
- `pipeline/`: contains the scripts called by the pipeline - **this is the only place for modules that can be run directly** (i.e. those with main guards). Some pipeline scripts use a mix of classes and adapters, which is not ideal but I did not have time to fix it.
- `aux_methods/`: contains the persistent logger config.

Other changes:
- Persistent logging config that logs both to file and console (the logging level for both file and console can be set in the file).
- Removed unneeded dependencies from pipeline
- Incremented pipeline version to 0.0.2 
- Scraped data folder is now a variable
- Changed `SimpleScraper` to just `Scraper`
- `Scraper` now propagates "label:" instead of "org_nr:"

### CODE REVIEW

File: adapters/scb.py [hunk 1]
```diff
@@ -288,6 +273,18 @@ def fetch_companies_from_db(self, sni_code, no_url=False):
...
         return list(companies)
 
+    def fetch_all_companies_from_db(self, no_url=False):
+        """
+        Fetch all companies from the database.
+        :param no_url: if True then only companies with missing urls will be returned.
+        returns a list of companies:
+        """
+        if no_url:
+            companies = self.mongo_client[Schema.DB][Schema.COMPANIES].find({"url": {"$regex": r"/\S"}})
```
Comment (simlar-0) [inline adapters/scb.py:283]: This fetches all companies that have a URL, so param should be has_url?

File: adapters/extract.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    Example usage:
+            ```
+            scb = SCBinterface()
+            scb.fetch_companies_from_api(
+                sni_start="00000",
+                sni_stop="02300", 
+                fetch_limit=10)
+            ```
+    """
```
Comment (simlar-0) [inline adapters/extract.py:18]: Update docstring

File: adapters/scrape.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    Example usage:
+            ```
+            scb = SCBinterface()
+            scb.fetch_companies_from_api(
+                sni_start="00000",
+                sni_stop="02300", 
+                fetch_limit=10)
+            ```
+    """
```
Comment (simlar-0) [inline adapters/scrape.py:18]: Update docstring

File: adapters/train.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    Example usage:
+            ```
+            scb = SCBinterface()
+            scb.fetch_companies_from_api(
+                sni_start="00000",
+                sni_stop="02300", 
+                fetch_limit=10)
+            ```
+    """
```
Comment (simlar-0) [inline adapters/train.py:19]: Update docstring

File: pipeline/divide_dataset.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import typer
+from pathlib import Path
+from typing_extensions import Annotated
+
+from adapters.train import TrainAdapter
+from adapters.extract import ExtractAdapter
+from adapters.scb import SCBAdapter
+
+def main(
+            percentage_training_split: Annotated[int, typer.Argument()] = 10
```
Comment (simlar-0) [inline pipeline/divide_dataset.py:18]: Default should be 70%

**Review by simlar-0 — CHANGES_REQUESTED** [review]
Great work! Some minor fixes

**Comment by yuvve** [comment]
> Great work! Some minor fixes

Thanks! I have fixed everything you mentioned and used that as an opportunity to improve the `no_url` / `has_url` parameters, so that they have now consistent behavior across both functions.

**Pull request merged**